codesign/test/codesign_test.dart

```diff
@@ -2,11 +2,64 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:codesign/codesign.dart';
-import 'package:test/test.dart';
+import 'package:codesign/codesign.dart' as cs;
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import './src/common.dart';
+
+/// A fake file visitor for testing purpose.
+///
+/// Upload for notarization is overriden, and the timer to check notarization
+/// status is fired instantly.
+class FakeCodesignVisitor extends cs.FileCodesignVisitor {
```

Comment:
umm yeah injection through constructor would work. The major reason I am proposing a fake override of class is that, the codesign visitor functions that will be overridden are not trivial to inject, like the use of [todo visitor](https://github.com/flutter/flutter/pull/101732/files#diff-9c1d5f7d9003b502c5198d32b4f5308f26153403050fc2557ff0674ab347494cR142) and [validation visitor](https://github.com/flutter/flutter/pull/101732/files#diff-9c1d5f7d9003b502c5198d32b4f5308f26153403050fc2557ff0674ab347494cR182) in flutter/conductor, and my previous implementation of zip vistor and [fake visitor](https://github.com/flutter/cocoon/pull/1861/files#diff-dc077581e9e225d7519e54fa1473c068ef326b02974cee6b30540a4a251a7881R127).

If we inject, I think we would inject many arguments through constructor, such as the implementation of function `notarize`, function `parse`, function `visit binary / zip / directory ` , `timer`, `zip / unzip` etc. , and my understanding is that the constructor will be bulky and not easy to initialize. umm I could be wrong, but I think a override would be clearer in our case?

Reply:
I have now changed all parameters to be passed through constructor. I am a bit concerned about passing in the `Function` fields, of which we want to override. For example, to override the `codesignVisitor.visitDirectory` function, I am currently passing the `visitDirectory` function as a parameter to `codesignVisitor` constructor. I think it might easier if we instantiate a `fakeVistior` class, and override the `visitorDirectory` function in the fake class?